stable/python/kubeless.py

```diff
@@ -49,8 +49,22 @@ def metrics():
     bottle.response.content_type = prom.CONTENT_TYPE_LATEST
     return prom.generate_latest(prom.REGISTRY)
 
+def enable_cors(fn):
+    def _enable_cors(*args, **kwargs):
+        # set CORS headers
+        bottle.response.headers['Access-Control-Allow-Origin'] = '*'
+        bottle.response.headers['Access-Control-Allow-Methods'] = 'GET, POST, PUT, DELETE, OPTIONS'
+        bottle.response.headers['Access-Control-Allow-Headers'] = 'Origin, Accept, Content-Type, X-Requested-With, X-CSRF-Token'
 
-@app.route('/<:re:.*>', method=['GET', 'POST', 'PATCH', 'DELETE'])
+        if bottle.request.method != 'OPTIONS':
```

Comment:
I would suggest to do the opposite, return CORS headers if the method is OPTIONS. We can mimic what is currently being done in the [nodejs runtime](https://github.com/kubeless/runtimes/blob/master/stable/nodejs/kubeless.js#L138).

Something like (untested code, please test this):

```
def enable_cors(req, fn):
    def _enable_cors(*args, **kwargs):
        # set CORS headers
        bottle.response.headers['Access-Control-Allow-Origin'] = '*'

        if bottle.request.method != 'OPTIONS':
            bottle.response.headers['Access-Control-Allow-Methods'] = req.get_header('access-control-request-method')
            bottle.response.headers['Access-Control-Allow-Headers'] = req.get_header('access-control-request-headers')
        else:
            # actual request; reply with the actual response
            return fn(*args, **kwargs)

    return _enable_cors
```

Note that we are allowing any method and header from the request, that way we avoid having to chose the base image if there is a new method/header that we want to allow.